### PR TITLE
set out routes and middleware for the user model for autorization

### PR DESCRIPTION
**Pull request by kathinka**
..autentication and logout. added a few optional improvements to the model, and routes.

connected login, logout, register with db.
added logout.
added routes i frontend

this is more than the routes task,  but i had to do it to test the routes for the api anyway.

if not ok, i can refrain from pushing more than the routes and middleware and logout for instance :)

### CODE REVIEW

File: backend/routes/routes.js [hunk 1]
```diff
@@ -47,25 +48,26 @@ router.post("/registration", async (req, res) => {
     if (existingUser) {
```
Comment (El1an3) [inline backend/routes/routes.js:48]: Good one, I would not have thought of this.

File: frontend/src/routes/Routes.jsx [hunk 2]
```diff
@@ -1,3 +1,23 @@
...
+import { Landing } from "../pages/Landing.jsx";
+import { Login } from "../pages/Login.jsx";
+import { Register } from "../pages/Register.jsx";
+import { Dashboard } from "../pages/Dashboard.jsx";
+import { Logout } from "../pages/Logout.jsx";
+
+
+export const AppRoutes = () => {
+  return <>
+    <Router>
```
Comment (El1an3) [inline frontend/src/routes/Routes.jsx:11]: I would have done more or less the same. 😉

**Review by El1an3 — APPROVED** [review]

File: backend/routes/routes.js (hunk 1 above)
Comment (kathinka) [reply backend/routes/routes.js:48]: ended up removing it when i saw that we only used email for logging in. but oh well, now we no how to use or operator in Mongodb :)

File: frontend/src/routes/Routes.jsx (hunk 2 above)
Comment (kathinka) [reply frontend/src/routes/Routes.jsx:11]: great minds think alike 😄

**Review by Yifan-858 — APPROVED** [review]

**Review by vittoriamatteoli — APPROVED** [review]

**Pull request merged**